extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -229,4 +234,44 @@ private function isEuropaTheme() {
 
 		return !empty( $wgEnablePortableInfoboxEuropaTheme );
 	}
+
+	private function createSmartGroups( $groupData, $rowCapacity ) {
+		$result = [ ];
+		$capacity = 0;
+		$smartGroup = [ ];
+
+		foreach ( $groupData as $item ) {
+			$data = $item['data'];
+
+			if ( $item['type'] === 'data' && ( !isset( $data['layout'] ) || $data['layout'] !== 'default' ) ) {
+
+				if ( $capacity + $data['span'] > $rowCapacity ) {
+					$result = array_merge( $result, $this->calculateSmartWidths( $smartGroup, $capacity ) );
```

Comment:
I'd extract these 3 lines to separated function to avoid code duplication as it's the same as few lines below